feat(rgpd): add matomo iframe

revolunet opened the pull request:
Ajout iframe Matomo dans politique de confidentialité

cc @ThomasMenant 

Par contre c'est pas très beau car l'iframe est complètement styleless

<img width="688" alt="Capture d’écran 2020-01-28 à 15 21 30" src="https://user-images.githubusercontent.com/124937/73272965-856b3a00-41e3-11ea-93a2-27cdae6d9ac3.png">

Commit:
feat(rgpd): add matomo iframe

UnbearableBear commented on packages/code-du-travail-frontend/pages/politique-confidentialite.js:
```diff
@@ -151,6 +151,16 @@ const CookiePolicy = ({ pageUrl, ogImage }) => {
             </p>
+            <p>
+              Pour l&apos;outil Matomo, vous pouvez décider de ne jamais être
+              suivi, y compris anonymement :
```
A un nbsp; près, c'était parfait !

UnbearableBear commented:
Ca commence à devenir dur dur les styles matomo / google analytics sur le site :'(

Commit:
add nbsp

Commit:
Update politique-confidentialite.test.js.snap

dependabot[bot] committed:
build(deps): bump tensorflow in /packages/code-du-travail-nlp (#2305)

Bumps [tensorflow](https://github.com/tensorflow/tensorflow) from 2.0.0 to 2.0.1.
- [Release notes](https://github.com/tensorflow/tensorflow/releases)
- [Changelog](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)
- [Commits](https://github.com/tensorflow/tensorflow/compare/v2.0.0...v2.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
fix: remove useless query escaping (#2304)

lionelB commented:
Pourquoi ne pas rajouter matomo dans tarte au citron ?

UnbearableBear commented:
On est pas supposés virer tarte au citron dans quelques semaines ?

lionelB commented:
Si Si :) mais il faudrait qu'on gère mieux l'optout de matomo (est ce qu'on peux le faire nous meme depuis le frontend ?)

ThomasMenant commented:
Accrochez-vous, en synthèse : 
- Tartocitron permet de gérer les cookies qui nécessitent le recueil du consentement de l'utilisateur (comme GA / Hotjar...) et configurer le fameux bandeau cookie ;
- Matomo fait partie des outils qui sont dispensés du recueil du consentement, en somme utiliser Matomo bien configuré permet de ne pas afficher de bandeau cookie ;

Partant de là, mettre Matomo dans Tartaucitron revient à s'imposer le fameux bandeau pour rien (si on suppose que le reste du produit ne comporte pas de cookies intrusifs). L'idée c'est aussi d'éviter de créer un précédent : _"le CDTN a mis Matomo dans le bandeau, on sait que c'est inutile mais c'est génial, tous les produits doivent faire pareil."_

Si on partage le même objectif qui est : être exemplaire en matière de vie privée, viser la suppression du bandeau (peu utile, peu esthétique), alors on privilégie les outils bien configurés et autohébergés.

Seul bémol, si le recueil du consentement n'est pas nécessaire pour Matomo, on est censés proposer une solution d'opt-out (la fameuse iframe), grosso modo c'est une fonctionnalité "do not track". En l’occurrence, c'est une demande expresse du ministère (je peux vous envoyer les mails).

Le compromis est de se dire, il vaut mieux avoir une iframe peu esthétique dans une page statique accessible depuis le footer, qu'un bandeau dégueu et inutile.

------
Maintenant si on applique ce raisonnement au CDTN en l'état vous avez la double peine :
1- bandeau cookie (tartaucitron) pour gérer le tracker GA
2- iframe opt-out pour Matomo

Je pense également que supprimer Tartaucitron dans quelques semaines (et évidemment les cookies concernés) est l'approche la plus pérenne.

Courage

lionelB commented:
@ThomasMenant Merci  pour la synthèse, ma proposition etait pas très clair j'avais plus en tete de mettre matomo dans tartaucitron durant la période d'acquisition du trafic, pour avoir le temps de trouver comment faire pour gérer l'iframe proprement mais bon... être pragmatique c'est bien aussi :)

Commit:
Merge branch 'matomo-iframe' of github.com:SocialGouv/code-du-travail-numerique into matomo-iframe

Commit:
test(frontend): update snaps

Commit:
feat(front): update iframe styles
